File: zs-leetcode-go/BestTimetoBuyandSellStock/BestTimetoBuyandSellStock.go
package BestTimetoBuyandSellStock

func _max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
func maxProfit(prices []int) int {
	if len(prices) == 0{
		return 0
	}
	max := 0
	for out := 0; out < len(prices)-1; out++ {
		for inner := out+1; inner < len(prices); inner++ {
			if prices[inner] > prices[out] {
				max = _max(prices[inner]-prices[out], max)
			}
		}
	}
	return max
}

File: sjq-leetcode-C++/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() == 0) return 0;
        while(prices.back() ==0) prices.pop_back();
        int i=0;
        int res=0;
        while(i<prices.size()-1){
            int j = i+1;
            while(j<prices.size()){
                res = max(res, prices[j]-prices[i]);
                j++;
            }
            i++;
        }
        return res;
    }
};
